oracle/src/main.rs

```diff
@@ -21,23 +21,45 @@ const NAME: &str = env!("CARGO_PKG_NAME");
 const ABOUT: &str = env!("CARGO_PKG_DESCRIPTION");
 
 const CONFIRMATION_TARGET: u32 = 1;
+
 const BTC_DECIMALS: u32 = 8;
-const DOT_DECIMALS: u32 = 10;
+const BTC_CURRENCY: &str = "btc";
+
+struct CurrencyIdArg(CurrencyId);
```

Comment:
I have a slight preference to use `CurrencyId` as arg directly, and to specify a parsing function. But doesn't have to block merging